### PR TITLE
Collect more data if CI encounters a failure

### PR DESCRIPTION
**Pull request by guillaumerose**
zip files with more data can be found here: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/code-ready_crc/1598/pull-ci-code-ready-crc-master-e2e-crc/1316460172016619520/artifacts/e2e-crc/gather-crc/test-results/

### CODE REVIEW

**Comment by guillaumerose** [comment]
Why openshift CI is not triggered ? oO

**Comment by guillaumerose** [comment]
it works: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/code-ready_crc/1598/pull-ci-code-ready-crc-master-e2e-crc/1316460172016619520/artifacts/e2e-crc/gather-crc/test-results/ !

**Comment by guillaumerose** [comment]
I can't login on ci.centos.org :/ If someone can verify the zip file is present, it would be great!

**Comment by guillaumerose** [comment]
Artifacts are here: http://artifacts.ci.centos.org/minishift/crc/pr/1598/

File: test/integration/crcsuite/collect.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+			Dir:    constants.CrcBaseDir,
+			Target: "tree.txt",
+		},
+		file(constants.ConfigPath),
+		file(constants.LogFilePath),
+		file(constants.DaemonLogFilePath),
+		file(filepath.Join(constants.MachineBaseDir, "machines", constants.DefaultName, "config.json")),
+		file("/etc/hosts"),
+		vmFile("/etc/resolv.conf"),
+		vmFile("/etc/hosts"),
```
Comment (cfergeau) [inline test/integration/crcsuite/collect.go:50]: `file("/etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf")` and `file("/etc/NetworkManager/dnsmasq.d/crc.conf")` would be useful as well. `journalctl` output on the host would be useful too, at least for NetworkManager logs. Maybe also `file("/var/log/libvirt/qemu/crc.log"`) (need to be root to read this).

File: test/integration/crcsuite/collect.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+		vmCommand("journalctl -n 1000 --no-pager", "journalctl.txt"),
+		vmCommand("ifconfig", "ifconfig.txt"),
+		vmCommand("dmesg", "dmesg.txt"),
```
Comment (cfergeau) [inline test/integration/crcsuite/collect.go:53]: isn't this redundant with `journalctl` just above?

File: test/integration/crcsuite/collect.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+		},
+		file(constants.ConfigPath),
+		file(constants.LogFilePath),
+		file(constants.DaemonLogFilePath),
+		file(filepath.Join(constants.MachineBaseDir, "machines", constants.DefaultName, "config.json")),
+		file("/etc/hosts"),
+		vmFile("/etc/resolv.conf"),
+		vmFile("/etc/hosts"),
+		vmCommand("journalctl -n 1000 --no-pager", "journalctl.txt"),
+		vmCommand("ifconfig", "ifconfig.txt"),
```
Comment (cfergeau) [inline test/integration/crcsuite/collect.go:52]: I think nowadays this has been superseded by `ip addr`

**Review by cfergeau — COMMENTED** [review]
Looking at one of the artifacts after a failed CI, the files seem to store the error that occurred as their content:
- `vm/resolv.conf`: `Cannot load machine: Machine "crc" does not exist. Use "crc start" to add a new one.`
- `config.json`: `open /home/crc_ci/.crc/machines/crc/config.json: no such file or directory`

My problem with this is that I don't get what would be causing this behaviour by looking at this PR :(

File: test/integration/crcsuite/collect.go (hunk 3 above)
Comment (guillaumerose) [reply test/integration/crcsuite/collect.go:52]: :+1:

File: test/integration/crcsuite/collect.go (hunk 1 above)
Comment (guillaumerose) [reply test/integration/crcsuite/collect.go:50]: Added apart the qemu logs as I dont have root for the moment.

File: test/integration/crcsuite/collect.go (hunk 2 above)
Comment (guillaumerose) [reply test/integration/crcsuite/collect.go:53]: Even if dmesg info are not in the first 1000 lines?

**Comment by guillaumerose** [comment]
Errors are not written in place of the file anymore now. I just want something simple enough to help for the CI. This is not yet for the end user.

File: test/integration/crcsuite/collect.go (hunk 2 above)
Comment (cfergeau) [reply test/integration/crcsuite/collect.go:58]: ah, why limit it to 1000 though? `journalctl --boot=-0 --no-pager`?

**Review by cfergeau — APPROVED** [review]

**Pull request merged**